build-system/babel-config/pre-closure-config.js

```diff
@@ -0,0 +1,110 @@
+        './polyfills/domtokenlist': ['installDOMTokenList'],
+        './polyfills/fetch': ['installFetch'],
+        './polyfills/math-sign': ['installMathSign'],
+        './polyfills/object-assign': ['installObjectAssign'],
+        './polyfills/object-values': ['installObjectValues'],
+        './polyfills/promise': ['installPromise'],
+        './polyfills/array-includes': ['installArrayIncludes'],
+        '../third_party/css-escape/css-escape': ['cssEscape'],
+        // Imports that are not needed for valid transformed documents.
+        '../build/ampshared.css': ['cssText', 'ampSharedCss'],
+        '../build/ampdoc.css': ['cssText', 'ampDocCss'],
+      },
+    },
+  ];
+  const reactJsxPlugin = [
+    '@babel/plugin-transform-react-jsx',
+    {
+      pragma: 'Preact.createElement',
+      pragmaFrag: 'Preact.Fragment',
+      useSpread: true,
+    },
+  ];
+
+  const preClosurePlugins = [
+    './build-system/babel-plugins/babel-plugin-transform-fix-leading-comments',
+    '@babel/plugin-transform-react-constant-elements',
+    reactJsxPlugin,
+    './build-system/babel-plugins/babel-plugin-transform-inline-configure-component',
+    // TODO(alanorozco): Remove `replaceCallArguments` once serving infra is up.
+    [
+      './build-system/babel-plugins/babel-plugin-transform-log-methods',
+      {replaceCallArguments: false},
+    ],
+    './build-system/babel-plugins/babel-plugin-transform-parenthesize-expression',
+    './build-system/babel-plugins/babel-plugin-is_minified-constant-transformer',
+    './build-system/babel-plugins/babel-plugin-transform-amp-extension-call',
+    './build-system/babel-plugins/babel-plugin-transform-html-template',
+    './build-system/babel-plugins/babel-plugin-transform-version-call',
+    './build-system/babel-plugins/babel-plugin-transform-simple-array-destructure',
+    replacePlugin,
+    argv.single_pass
+      ? './build-system/babel-plugins/babel-plugin-transform-amp-asserts'
+      : null,
+    argv.esm ? filterImportsPlugin : null,
+    argv.esm
+      ? './build-system/babel-plugins/babel-plugin-transform-function-declarations'
+      : null,
+    isCheckTypes
+      ? './build-system/babel-plugins/babel-plugin-transform-simple-object-destructure'
+      : './build-system/babel-plugins/babel-plugin-transform-json-configuration',
+    !(argv.fortesting || isCheckTypes)
+      ? [
+          './build-system/babel-plugins/babel-plugin-amp-mode-transformer',
+          {isEsmBuild: argv.esm},
+        ]
+      : null,
+    !(argv.fortesting || isCheckTypes)
+      ? './build-system/babel-plugins/babel-plugin-is_dev-constant-transformer'
+      : null,
+    argv.single_pass ? 'transform-es2015-modules-commonjs' : null,
+  ].filter(Boolean);
+  const preClosureConfig = {
+    compact: false,
+    ignore: devDependencies,
+    plugins: preClosurePlugins,
+    retainLines: true,
+  };
+  if (argv.esm) {
+    preClosureConfig.sourceType = 'module';
```

Comment:
Nit: Instead of a mutation after creation, you could use an alternate value for `sourceType` when not in `esm` mode. The default is 'module' though.

https://babeljs.io/docs/en/options#sourcetype

Reply:
Good observation. I've removed all `sourceType: 'module'` specifications since the `sourceType` isn't being overridden anywhere in our build code. Will follow up by re-testing all commands to make sure this is okay.